sockeye/quantization.py

```diff
@@ -0,0 +1,170 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not
+# use this file except in compliance with the License. A copy of the License
+# is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+import mxnet as mx
+from . import constants as C
+from mxnet.gluon.nn.activations import Activation
+import logging
+
+logger = logging.getLogger(__name__)
+
+# Modified from the source to mxnet.gluon.nn.basic_layers.Dense which is:
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+class QuantizableDense(mx.gluon.HybridBlock):
```

Comment:
Couldn't you inherit from `mx.gluon.nn.basic_layers.Dense` directly and only overwrite `cast()` and `hybrid_forward`?

Reply:
I agree. Tried to do this, will need a consultation with a gluon expert.

Reply:
what was the issue?

Reply:
I guess we need to carefully set the prefix for the inheriting class to make sure the parameter names match.